Cargo.toml

```diff
@@ -96,7 +96,7 @@ clap_lex = { path = "./clap_lex", version = "0.3.0" }
 bitflags = "1.2"
 unicase = { version = "2.6", optional = true }
 strsim = { version = "0.10",  optional = true }
-atty = { version = "0.2",  optional = true }
+is-terminal = { version = "0.3",  optional = true }
```

Comment:
I've reached out to the `atty` author to verify.

`is-terminal` looks a bit suspicious in that they do not provide any audit trail of how the crate changed since they forked from atty.  The commit history starts after they started making changes.

Reply:
Reaching out to the author sounds like a good call, yes.

As `is-terminal` being suspicous: I disagree. From what I can tell, they haven't *forked* from `atty`, but started from scratch, copying over only the heuristics needed on Windows from `atty`. With that approach in mind, I don't think it's suspicious, especially considering how both crates are small enough to fully read the source code in half an hour. I've completely compared the two implementations, and while there are a few differences, they come down to switching from `winapi` to `windows-sys` and to https://github.com/softprops/atty/pull/51 being the base for the copying.

Aside of that, the author of `is-terminal` has gotten their stuff merged upstream into Rust. This goes both for bigger own contributions like `io-lifetimes` (see https://github.com/rust-lang/rfcs/pull/3128 and https://github.com/rust-lang/rust/pull/87329) as well as just a bunch of smaller PRs. This of course doesn't mean that `is-terminal` has to be any good, but it's still a good sign IMO.

Reply:
If there really is a legit soundness issue with `atty` then that issue should be properly assessed and translated into evaluated RustSec advisory - if any - with `informational = "unsound"` to let everyone know properly about it.

Please feel free to raise issue about `atty` in RustSec and we can deal with it's status there or I can check on it later -

The issue in the subject starts with "Potential" and proving and documenting the actual soundness of it is left for someone else to do where nobody - op or anyone else - sadly has properly evaluated / acted on it leaving it as academic guessing.

The op back in the time seems to have abandoned the issue on trying to iterate on it for validity ?

I wouldn't be going replacing a long running crate just by guessing alone -

We also got nothing pre-existing on `atty` yet either in the database so doing nothing here would be the best option for now.

If `atty` indeed would be unsound then we can act on it and coordinate properly but we have to validate the potential issue properly first to see what the issue is and what the impact may or may not be.

Missing audit trail indeed on `is-terminal` is concerning and I personally would not switch to a direct fork without it either.

btw however `is-terminal` seems maintained by @sunfishcode who is recent libs team addition and wasmtime maintainer among a lot of other things they maintain so the maintainer is definitely trustworthy :)

Nonetheless I would not be replacing this crate necessarily as of yet if at all -

Perhaps @sunfishcode could give some assurances re: what commits have been applied on top of `atty` ?

Reply:
> If there really is a legit soundness issue with `atty` then that issue should be properly assessed and translated into evaluated RustSec advisory - if any - with `informational = "unsound"` to let everyone know properly about it.
> 
> Please feel free to raise issue about `atty` in RustSec and we can deal with it's status there or I can check on it later -

Fair enough, that's the route I will take next time I stumble over an issue like this. Not sure why I didn't think of that tbh..

> The issue in the subject starts with "Potential" and proving and documenting the actual soundness of it is left for someone else to do where nobody - op or anyone else - sadly has properly evaluated / acted on it leading to guessing.
> 
> The op back in the time seems to have abandoned the issue on trying to iterate on it for validity?

I'm not really happy about how clearly documented this either, but my reading of the PR description (which is from the same OP who also opened the issue, merely 2.5h later) is that the unsoundness was confirmed: "Fixes https://github.com/softprops/atty/issues/50 as the problem was confirmed by a user on the community discord". A more detailed explanation or maybe at least a link to the specific message or thread on discord would have been nice, but is not available to my knowledge. Maybe someone from RustSec can look into this a bit deeper and decide whether it's actually a problem or not?

Reply:
> From what I can tell, they haven't forked from atty, but started from scratch, copying over only the heuristics needed on Windows from atty

To clarify, I was going off of softprops being listed as one of the authors.

Reply:
It's been a month, I assume that there's not been a reaction from the `atty` author yet? From the other perspective: I've just reached out to the rustsec mail address, hoping that someone from there can either confirm the problem or confirm that it's not a problem.

Reply:
> Delivery Status Notification (Failure)

Or well, I've tried to reach out to the rustsec mail address.

@pinkforest as you've suggested that this should go through rustsec, what would you suggest as the proper way to get the attention from someone who can then evaluate whether it's a problem or not?

Reply:
I think this would happen at https://github.com/rustsec/advisory-db

Reply:
We have a mail address? :) Anyways I've flagged https://github.com/softprops/atty/issues/57 to ping @softprops about it.
Maintainer really should have the say on this and they might be on just some hiatus.

Reply:
There's one listed here: https://github.com/rustsec/advisory-db/blob/main/CONTRIBUTING.md#faq. You don't seem to "have" a mail address though, considering google is telling me off when trying to send stuff there.

Thanks for opening the issue upstream, worded way better than I could've done it. Totally agree that maintainers should be in on conversations like this, would've been better to start with more comments upstream instead of going downstream first. I'll try to do better in the future :)

Reply:
I filed an issue for `atty` it on advisory-db. It looks indeed like a (theoretical) soundness issue, but I guess it is not an issue as long as you use a sane global allocator (?).

Reply:
It's indeed soundness issue and further investigating at Wg-unsafe-guideline
https://rust-lang.zulipchat.com/#narrow/stream/136281-t-lang.2Fwg-unsafe-code-guidelines/topic/.60.23.5Bglobal_allocator.5D.60.20academic.20dilemma/near/311530432

I was pointed to: https://github.com/rust-lang/rust/pull/98033#discussion_r958409731

Which was also found when the said code was being ported to std IsTerminal

The advisory has been merged to reflect the facts - thanks @niluxv and @jcgruenhage for nudging this over

Reply:
I did not now that this was being ported to std, but sounds like a useful thing to have in there. Either way, thanks @pinkforest for following up here!

Regarding this PR then, @epage is this something you'd be willing to merge until `IsTerminal` in std is stabilized and the MSRV is high enough to migrate to that?

Reply:
is-terminal maintainer here; sorry I missed the ping above.

The differences between is-terminal and atty are 
 - applied https://github.com/softprops/atty/pull/51 
 - applied https://github.com/softprops/atty/pull/54
 - changed from taking an enum argument to a handle argument

I made the mistake of not preserving this history, though I've reconstructed the original atty plus the two patches here:
 - https://github.com/sunfishcode/is-terminal/tree/history